scapy/contrib/isotp.py

```diff
@@ -0,0 +1,345 @@
+        # };
+
+    def __build_can_isotp_fc_options(self,
+                                     bs=CAN_ISOTP_DEFAULT_RECV_BS,
+                                     stmin=CAN_ISOTP_DEFAULT_RECV_STMIN,
+                                     wftmax=CAN_ISOTP_DEFAULT_RECV_WFTMAX):
+        return struct.pack(self.can_isotp_fc_options_fmt,
+                           bs,
+                           stmin,
+                           wftmax)
+        # == Must use native not standard types for packing ==
+        # struct can_isotp_fc_options {
+        #
+        # __u8  bs;               /* blocksize provided in FC frame       */
+        #                         /* __u8 value : blocksize. 0 = off      */
+        #
+        # __u8  stmin;            /* separation time provided in FC frame */
+        #                         /* __u8 value :                         */
+        #                         /* 0x00 - 0x7F : 0 - 127 ms             */
+        #                         /* 0x80 - 0xF0 : reserved               */
+        #                         /* 0xF1 - 0xF9 : 100 us - 900 us        */
+        #                         /* 0xFA - 0xFF : reserved               */
+        #
+        # __u8  wftmax;           /* max. number of wait frame transmiss. */
+        #                         /* __u8 value : 0 = omit FC N_PDU WT    */
+        # };
+
+    def __build_can_isotp_ll_options(self,
+                                     mtu=CAN_ISOTP_DEFAULT_LL_MTU,
+                                     tx_dl=CAN_ISOTP_DEFAULT_LL_TX_DL,
+                                     tx_flags=CAN_ISOTP_DEFAULT_LL_TX_FLAGS):
+        return struct.pack(self.can_isotp_ll_options_fmt,
+                           mtu,
+                           tx_dl,
+                           tx_flags)
+
+        # == Must use native not standard types for packing ==
+        # struct can_isotp_ll_options {
+        #
+        # __u8  mtu;              /* generated & accepted CAN frame type  */
+        #                         /* __u8 value :                         */
+        #                         /* CAN_MTU   (16) -> standard CAN 2.0   */
+        #                         /* CANFD_MTU (72) -> CAN FD frame       */
+        #
+        # __u8  tx_dl;            /* tx link layer data length in bytes   */
+        #                         /* (configured maximum payload length)  */
+        #                         /* __u8 value : 8,12,16,20,24,32,48,64  */
+        #                         /* => rx path supports all LL_DL values */
+        #
+        # __u8  tx_flags;         /* set into struct canfd_frame.flags    */
+        #                         /* at frame creation: e.g. CANFD_BRS    */
+        #                         /* Obsolete when the BRS flag is fixed  */
+        #                         /* by the CAN netdriver configuration   */
+        # };
+
+    def __getSockIfreq(self, sock, iface):
```

Comment:
Can you rename this method as `__get_sock_ifreq()`? That would be more consistent with the other methods.

Reply:
Done